sda-commons-shared-otel-tracing-example/src/test/java/org/sdase/shared/otel/tracing/example/OtelTracingApplicationIT.java

```diff
@@ -0,0 +1,92 @@
+package org.sdase.shared.otel.tracing.example;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.dropwizard.Configuration;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.sdk.testing.junit5.OpenTelemetryExtension;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.List;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.assertj.core.groups.Tuple;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class OtelTracingApplicationIT {
+  @RegisterExtension
+  private static final DropwizardAppExtension<Configuration> APP =
+      new DropwizardAppExtension<>(OtelTracingApplication.class, new Configuration());
+
+  @RegisterExtension
+  static final OpenTelemetryExtension otelTesting = OpenTelemetryExtension.create();
+
+  @BeforeEach
+  void setUp() throws IllegalAccessException {
+    // use InMemory Exporter for testing
+    FieldUtils.writeField(
+        APP.getApplication(),
+        "tracer",
+        otelTesting.getOpenTelemetry().getTracer(OtelTracingApplicationIT.class.getName()),
+        true);
```

Comment:
Can you find a better solution that does not use reflection to be ready for Java 17? What about providing a simple setter?